pkg/codegen/python/gen.go

```diff
@@ -2990,16 +2969,267 @@ func GeneratePackage(tool string, pkg *schema.Package, extraFiles map[string][]b
 	}
 	files.Add(filepath.Join(pkgName, "pulumi-plugin.json"), plugin)
 
-	// Finally emit the package metadata (setup.py).
+	// Next, emit the package metadata (setup.py).
 	setup, err := genPackageMetadata(tool, pkg, pkgName, info.Requires, info.PythonRequires)
 	if err != nil {
 		return nil, err
 	}
 	files.Add("setup.py", []byte(setup))
 
+	// Finally, if pyproject.toml generation is enabled, generate
+	// this file and emit it as well.
+	if info.PyProject.Enabled {
+		project, err := genPyprojectTOML(
+			tool, pkg, pkgName,
+		)
+		if err != nil {
+			return nil, err
+		}
+		files.Add("pyproject.toml", []byte(project))
+	}
+
 	return files, nil
 }
 
+func genPyprojectTOML(tool string,
+	pkg *schema.Package,
+	pyPkgName string,
+) (string, error) {
+	// First, create a Writer for everything in pyproject.toml
+	w := &bytes.Buffer{}
+	// Create an empty toml manifest.
+	schema := new(PyprojectSchema)
+	schema.Project = new(Project)
+	// Populate the fields.
+	schema.Project.Name = &pyPkgName
+	schema.Project.Keywords = pkg.Keywords
+
+	// Setting dependencies fails if the deps we provide specify
+	// an invalid Pulumi package version as a dep.
+
+	err := setDependencies(schema, pkg)
+	if err != nil {
+		return "", err
+	}
+
+	// These values are hardcoded to Pulumi's contact information.
+	// TODO(@Robbie): Third-party provider users can override this field.
```

Comment:
I intend to make a follow-up issue to allow overrides for these fields. That way, third-party providers can provide their contact information in the provider schema instead of our information.

Reply:
We already have "publisher" on the schema. I suspect we just need like "publisherEmail", we could probably use that in the other SDK generators as well.